### ISSUE TITLE
Docs on how to modify query in milestones?

### ISSUE THREAD

**Issue by magwo** [dchester/epilogue#170]
Hi,

I would like to prevent users, belonging to different organisations, from being able to see data from organisations other than the one they belong to. This spans a lot of models in the system so I would prefer to make a catch-all solution to make sure the organisation id is always enforced to be the user's organisation id.

I'm not able to deduce from the milestone docs on how to add this "filtering".

Basically my idea is to, in a milestone handler in ALL resources, add this: `where: { organisationId: user.organisationId }`

Is this a crazy approach? How else could you do it?

The goal is to strictly and mistake-proof separate organisations' data from each other.

**Comment by magwo** [comment]
I might have figured it out myself.. is this the correct/sane way? It appears to work:
```
context.criteria = context.criteria || {};
context.criteria['organizationId'] = res.locals.user.organization_id;
```

It would be great if the docs on milestone explained the criteria object a little more.

**Comment by mbroadst** [comment]
@magwo yep you figured it out. I agree, there should be more information about this in the documentation, would you be willing to contribute some documentation in a PR?

**Comment by magwo** [comment]
Ok thanks. Are there any caveats or limitations to the way I use the criteria object there? Are there any potential security problems where the "enforcement" might fail in spite of the criteria?

**Comment by mbroadst** [comment]
@magwo the idea behind epilogue is really very simple, there is a pretty flexible milestone system and a bunch of default implementations provided for auto generating restful routes for a resource. I'm not sure what controller you're working with, but let's use `READ` as an example:

The `ReadController` only implements the `fetch` action, and its criteria is initially defined here: https://github.com/dchester/epilogue/blob/master/lib/Controllers/read.js#L21. As you can see, whatever was previously defined in the `context.criteria` will be used if it exists, otherwise it will be defined as an empty object. Subsequently the criteria is copied to the `options` passed along to sequelize here: https://github.com/dchester/epilogue/blob/master/lib/Controllers/read.js#L35.

So as far as epilogue/sequelize is concerned, there aren't really any caveats here - it's pretty straightforward.

**Comment by magwo** [comment]
Ok cool.

**Comment by magwo** [comment]
What is the reason for the criteria object being called `criteria` rather than `where`? It seems `where` would be more in line with the Sequelize terminology - it appears to be the same object since it is set to the where object in the options object, right?

**Comment by mbroadst** [comment]
`criteria` is more semantically accurate, where `where` is just a SQL-ism. I imagine for that reason it was chosen for readability.

**Comment by magwo** [comment]
Ok I see, thanks. Does the docs contain any example of using the criteria object? Otherwise I might add that in a PR if I find the time.

**Comment by mbroadst** [comment]
@magwo the documentation is as available to you as it is to me 😄 from a cursory grep I couldn't see any use of it in the documentation aside from an example for caching.

**Comment by mbroadst** [comment]
@magwo I think we can close this now?

**Comment by magwo** [comment]
@mbroadst Regarding caveats, I discovered one, reported here:
https://github.com/dchester/epilogue/issues/171

Should be an easy fix, right? Quite important IMO since it prevents data security, or at least easily creates loopholes for an unsuspecting developer.
